### PR TITLE
Clean code

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Continue removing some redundant code

### CODE REVIEW

File: pkg/controllers/job/job_controller.go [hunk 1]
```diff
@@ -127,53 +122,18 @@ func NewJobController(config *rest.Config) *Controller {
...
-			default:
-				runtime.HandleError(fmt.Errorf("unable to handle object %T", obj))
-				return false
-			}
-		},
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc: cc.addCommand,
-		},
+	cc.cmdInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: cc.addCommand,
```
Comment (TommyLike) [inline pkg/controllers/job/job_controller.go]: Can we keep the ``ControllerdBy`` filter logic here?

File: pkg/controllers/job/job_controller.go [hunk 2]
```diff
@@ -127,53 +122,18 @@ func NewJobController(config *rest.Config) *Controller {
...
-				return false
-			}
-		},
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    cc.addPod,
-			UpdateFunc: cc.updatePod,
-			DeleteFunc: cc.deletePod,
-		},
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    cc.addPod,
```
Comment (TommyLike) [inline pkg/controllers/job/job_controller.go]: Ditto

File: pkg/controllers/job/job_controller_actions.go [hunk 3]
```diff
@@ -66,38 +66,20 @@ func (cc *Controller) killJob(jobInfo *apis.JobInfo, nextState state.NextStateFn
...
 
-			switch pod.Status.Phase {
-			case v1.PodRunning:
-				err := cc.deleteJobPod(job.Name, pod)
-				if err != nil {
-					running++
-					errs = append(errs, err)
-					continue
-				}
+			if err := cc.deleteJobPod(job.Name, pod); err != nil {
```
Comment (TommyLike) [inline pkg/controllers/job/job_controller_actions.go]: err == nil

File: pkg/controllers/job/job_controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller.go]: Is there any command resource not managed by volcano?

File: pkg/controllers/job/job_controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller.go]: This is not needed,  we already have many other checks in event handlers, like this

```
	taskName, found := pod.Annotations[vkbatchv1.TaskSpecKey]
	if !found {
		glog.Infof("Failed to find taskName of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}

	jobName, found := pod.Annotations[vkbatchv1.JobNameKey]
	if !found {
		glog.Infof("Failed to find jobName of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}

	version, found := pod.Annotations[vkbatchv1.JobVersion]
	if !found {
		glog.Infof("Failed to find jobVersion of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}
```

File: pkg/controllers/job/job_controller_actions.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_actions.go]: good catch

File: pkg/controllers/job/job_controller.go (hunk 1 above)
Comment (TommyLike) [reply pkg/controllers/job/job_controller.go]: which is faster? Just asking.

File: pkg/controllers/job/job_controller.go (hunk 2 above)
Comment (TommyLike) [reply pkg/controllers/job/job_controller.go]: Which is faster, just asking.

File: pkg/controllers/job/job_controller.go (hunk 1 above)
Comment (TommyLike) [reply pkg/controllers/job/job_controller.go]: No for now.

File: pkg/controllers/job/job_controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller.go]: The above check is inevitable, so it's enough, if we check twice, then absolutely double time wasting

**Pull request merged**